rust/email/src/lib.rs

```diff
@@ -86,7 +102,12 @@ fn c_char_to_string(str_: *const c_char) -> Result<String, i32> {
 #[no_mangle]
 pub extern "C" fn send_email(email_info: EmailInfo) -> i32 {
 	let from_str = c_char_to_string(email_info.from).unwrap();
-	let body_str = c_char_to_string(email_info.body).unwrap();
+	let filepath_str =
+		c_char_to_string(email_info.filepath).unwrap();
```

Comment:
Last one. Let's print the error to the user if we receive one.

Reply:
We want to print the error in the `send_mail()` function before the `return -1`, right?

Reply:
Actually, let's return something else there, not -1. So, if something went wrong with the file, print the error and return -2.

Reply:
Good idea